stl/inc/system_error

```diff
@@ -131,6 +131,11 @@ protected:
     public:
         constexpr explicit _Addr_storage(const uintptr_t _Addr_num) noexcept : _Num(_Addr_num) {}
         constexpr explicit _Addr_storage(error_category* const _Addr_ptr) noexcept : _Ptr(_Addr_ptr) {}
+
+        // TRANSITION: As of Boost 1.80.0, boost::system::detail::std_category assigns to _Addr.
+        constexpr void operator=(const uintptr_t _Addr_num) noexcept {
+            _Num = _Addr_num;

```

Comment:
I can't tolerate an assignment that doesn't return `*this`, even one that no one should use:
```suggestion
        constexpr _Addr_storage& operator=(const uintptr_t _Addr_num) noexcept {
            _Num = _Addr_num;
            return *this;
```

Reply:
Fair enough, I'll change it :joy_cat: